src/docs/guides/assets.md

```diff
@@ -0,0 +1,251 @@
+
+For each instance of the Stencil runtime that is loaded, there is a single asset base path.
+Oftentimes, this means there is only one asset base path per application using Stencil.
+
+## Resolution Overview
+
+The process of resolving an asset involves asking Stencil to build a path to the asset on the filesystem. 
+
+When an asset's path is built, the resolution is always done in a project's compiled output, not the directory containing the original source code.
+
+The example below uses the output of the [`www` output target](/docs/www) to demonstrate how assets are resolved.
+Although the example use the output of `www` builds, the general principle of how an asset is found holds for all output targets.
+
+When using the `www` output target, a `build/` directory is automatically created and set as the asset base path.
+An example `build/` directory and the assets it contains can be found below.
+
+```
+www/
+├── build/
+│   ├── assets/
+│   │   ├── logo.png
+│   │   └── scenery/
+│   │       ├── beach.png
+│   │       └── sunset.png
+│   └── other-assets/
+│       └── font.tiff
+└── ...
+```
+
+To resolve the path to an asset, Stencil's [`getAssetPath()` API](/docs/assets#getassetpath) may be used.
+When using `getAssetPath`, the assets in the directory structure above are resolved relative to `build/`.
+
+The code sample below demonstrates the return value of `getAssetPath` for different `path` arguments.
+The return value is a path that Stencil has built to retrieve the asset on the filesystem.
+```ts
+// with an asset base path of "/build/":
+
+// '/build/assets/logo.png'
+getAssetPath('assets/logo.png');
+// '/build/assets/scenery/beach.png'
+getAssetPath('assets/scenery/beach.png');
+// '/build/other-assets/font.tiff'
+getAssetPath('other-assets/font.tiff'); 
+```
+
+## Making Assets Available
+
+In order to be able to find assets at runtime, they need to be found on the filesystem from the output of a Stencil build.
+In other words, we need to ensure they exist in the distribution directory.
+This section describes how to make assets available under the asset base path.
+
+### assetsDirs
+
+The `@Component` decorator can be [configured with the `assetsDirs` option](/docs/component#component-options). 
+`assetsDirs` takes an array of strings, where each entry is a relative path from the component to a directory containing the assets the component requires.
+
+When using the `dist` or `www` output targets, setting `assetsDirs` instructs Stencil to copy that folder into the distribution folder.
+When using other output targets, Stencil will not copy assets into the distribution folder.
+
+Below is an example project's directory structure containing an example component and an assets directory.
+
+```
+src/
+└── components/
+    ├── assets/
+    │   ├── beach.jpg
+    │   └── sunset.jpg
+    └── stencil-asset.tsx
+```
+
+Below, the `stencil-asset` component will correctly load the assets based on it's `image` prop.
```

Comment:
Totally get what you're going for here, but could we call this component something different? Maybe just `my-component`? Though I think most will understand this, I fear that `stencil-asset` may suggest to some that the component is an asset itself.

Reply:
👍 Done in fb492c5d0..8c9b799d7